### PR TITLE
Remove always true cases in wallet/crypter.cpp encrypt and decrypt function

### PR DESCRIPTION
**Pull request by JeremyRubin**

### CODE REVIEW

File: src/wallet/crypter.cpp [hunk 1]
```diff
@@ -59,11 +59,9 @@ bool CCrypter::Encrypt(const CKeyingMaterial& vchPlaintext, std::vector<unsigned
     vchCiphertext = std::vector<unsigned char> (nCLen);
 
     EVP_CIPHER_CTX ctx;
-
-    bool fOk = true;
-
     EVP_CIPHER_CTX_init(&ctx);
-    if (fOk) fOk = EVP_EncryptInit_ex(&ctx, EVP_aes_256_cbc(), NULL, chKey, chIV) != 0;
+
+    bool fOk = EVP_EncryptInit_ex(&ctx, EVP_aes_256_cbc(), NULL, chKey, chIV) != 0;
```
Comment (jonasschnelli) [inline src/wallet/crypter.cpp]: IMO this results in the same (remove code) as above.
The L66 EVP function only gets executed when fOk is true (which is always true at L66). fOk will then be set to true if the EVP function did not return 0.

I think the current code works right.

Comment (paveljanik) [inline src/wallet/crypter.cpp]: ... and looks ugly 8)

ACK

**Comment by jonasschnelli** [comment]
Yes. This code is cleaner.

utACK.

**Comment by theuni** [comment]
Change looks fine, but we're about to nuke it anyway with #5949. Would prefer to avoid the rebase.

**Comment by sipa** [comment]
No strong opinion about the code change, but agree with @theuni.

**Comment by MarcoFalke** [comment]
NACK, I like the previous code better because the "formatting" is uniform. Also, I am pretty sure the compiler will remove this case. Apart from that, agree with @theuni

**Comment by laanwj** [comment]
Yes, let's focus on #5949 instead.

**Pull request closed without merging**